packages/ketcher-core/src/application/render/restruct/resgroup.ts

```diff
@@ -234,18 +237,97 @@ class ReSGroup extends ReObject {
             tfx(b0.y),
           )
           .attr(options.hoverStyle);
+        otherHovers.push(sGroupItem.hovering);
       }
-      set.push(sGroupItem.hovering);
 
       SGroup.getAtoms(render.ctab.molecule, sGroupItem).forEach((aid) => {
         const atom = render?.ctab?.atoms?.get(aid);
 
-        set.push(atom?.makeHoverPlate(render));
+        hoversToCombine.push(atom?.makeHoverPlate(render));
       }, this);
       SGroup.getBonds(render.ctab.molecule, sGroupItem).forEach((bid) => {
-        set.push(render?.ctab?.bonds?.get(bid)?.makeHoverPlate(render));
+        hoversToCombine.push(
+          render?.ctab?.bonds?.get(bid)?.makeHoverPlate(render),
+        );
       }, this);
-      render.ctab.addReObjectPath(LayerMap.hovering, this.visel, set);
+
+      const elements: Element[] = [];
+
+      hoversToCombine.forEach((item) => {
+        if (item?.node) {
+          elements.push(item.node);
+          item.node.remove();
+        }
+      });
+
+      paperjs.setup(document.createElement('canvas')); // Paper.js works on an offscreen canvas
+
+      // Helper function to convert SVG elements into Paper.js paths
+      // eslint-disable-next-line no-inner-declarations
+      function paperPathFromSVGElement(element) {
```

Comment:
It seems like some common purpose util function, perhaps just declare and export it outside of this function/file? This will also help to avoid eslint suppressing above